### DISCUSSION TITLE
Configure PILOS to be itself Load-Balanced

### DISCUSSION THREAD

**Discussion by lakano** [THM-Health/PILOS#488] [General]
Hi !

PILOS already support natively the load-balancing between BBB servers, but what happening if PILOS itself had an outage?
We could uses a Load-Balancer in front of at least 2 PILOS servers on different provider/region, but we should make a list of all the necessary changes in PILOS to eventually add this feature.

- We should be able to sync the MariaDB with Galera in master-master.
- I'm not sure, may be we also need to sync the storage/ folder between all PILOS instances (syncthing?)
- All the command tasks in schedule:run should be idempotent (to prevent duplicates in DB)

Did you already explore this possibility? Do you aware about others restrictions that need to adjusts?

Best

**Comment by samuelwei** [comment]
Hi @lakano,

in v4 we have working on the scalabilty of PILOS. Please have a look at our docs: https://thm-health.github.io/PILOS/docs/administration/advanced/scaling#horizontal-scaling

We cannot provide guides for how to setup a HA Database, Storage, Redis, etc. but PILOS has been optimised to work in such scenarios.
You need to make sure, that all containers use the same storage, database and redis

**Reply by lakano** [reply]
This is a HUGE feature. 
For the storage, I don't known if there is open-source solutions that permit to uses distant and different datacenter ( to mitigate any provider outage ), but may be syncthing could be uses.
About MariaDB / Redis, I known there is HA solutions, but I need to investigate how to do that.
Thank you for this amazing feature!
